rust/src/datatypes.rs

```diff
@@ -280,6 +306,54 @@ impl Field {
         }
         None
     }
+
+    #[async_recursion]
+    async fn load_dictionary<'a>(&mut self, reader: &'a ObjectReader<'_>) -> Result<()> {
+        if let DataType::Dictionary(_, value_type) = self.data_type() {
+            assert!(self.dictionary.is_some());
```

Comment:
do we want to panic here or just return Err?

Reply:
feel this should never happen? i.e,. the data type and fields are not consistent. it is a bug instead of error, which could be handled by users